### PR TITLE
[DEPENDS ON #2] Add waterpoint markers

### PR DESCRIPTION
**Pull request by uniphil**
Note: this branch was split off the data branch, which is not merged yet (see #2 ). It should be merged before this one.

### CODE REVIEW

File: app/scripts/components/leaflet/waterpoint-marker.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    WATER_POINT_NAME: PropTypes.string.isRequired,
+    children: PropTypes.node,
+    map: PropTypes.instanceOf(Map),  // injectd by <Map>
+  },
+  render() {
+    const { children, map, LATITUDE, LONGITUDE } = this.props;
+    const clonedChildren = React.Children.map(children, child =>
+      child ? React.cloneElement(child, { map }) : null);
+    return (
+      <Marker map={this.props.map} position={[LATITUDE, LONGITUDE]}>
```
Comment (esoliani) [inline app/scripts/components/leaflet/waterpoint-marker.jsx]: remove this.props. from the map

File: app/scripts/components/leaflet/waterpoint-marker.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        <Popup map={this.props.map}>
```
Comment (esoliani) [inline app/scripts/components/leaflet/waterpoint-marker.jsx]: remove this.props. from the map

File: app/scripts/components/leaflet/waterpoint-marker.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    LATITUDE: PropTypes.number.isRequired,
+    LONGITUDE: PropTypes.number.isRequired,
+    WATER_POINT_CODE: PropTypes.string.isRequired,
+    WATER_POINT_NAME: PropTypes.string.isRequired,
+    children: PropTypes.node,
+    map: PropTypes.instanceOf(Map),  // injectd by <Map>
+  },
+  render() {
+    const { children, map, LATITUDE, LONGITUDE } = this.props;
+    const clonedChildren = React.Children.map(children, child =>
```
Comment (esoliani) [inline app/scripts/components/leaflet/waterpoint-marker.jsx]: why do we need to clone clildren?

File: app/scripts/components/leaflet/waterpoint-marker.jsx (hunk 2 above)
Comment (uniphil) [inline app/scripts/components/leaflet/waterpoint-marker.jsx]: fixed

File: app/scripts/components/leaflet/waterpoint-marker.jsx (hunk 3 above)
Comment (uniphil) [inline app/scripts/components/leaflet/waterpoint-marker.jsx]: In this case we don't, so I took it out.

If we had children that could be react-leaflet components, then we would need to clone them to ensure they have the `map` prop.

**Pull request merged**